analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -0,0 +1,200 @@
+            repoConfig: RepositoryConfiguration
+        ) = SpdxDocumentFile(managerName, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    private fun SpdxPackage.toIdentifier() =
+        Identifier(
+            type = managerName,
+            namespace = originator.extractOrganization().orEmpty(),
+            name = name,
+            version = versionInfo
+        )
+
+    private fun String.extractOrganization() =
+        lineSequence().mapNotNull { line ->
+            line.removePrefix("Organization: ").takeIf { it != line }
+        }.firstOrNull()
+
+    private fun String.mapNotPresentToEmpty() = takeUnless { SpdxConstants.isNotPresent(it) }.orEmpty()
+
+    /**
+     * Return the dependencies of [pkg] defined in [doc] of the [SpdxRelationship.Type.DEPENDENCY_OF] type.
+     */
+    private fun getDependencies(pkg: SpdxPackage, doc: SpdxDocument): SortedSet<PackageReference> =
+        getDependencies(pkg, doc, SpdxRelationship.Type.DEPENDENCY_OF) { target ->
+            val dependency = doc.packages.single { it.spdxId == target }
+            PackageReference(
+                id = dependency.toIdentifier(),
+                dependencies = getDependencies(dependency, doc)
+            )
+        }
+
+    /**
+     * Return the dependencies of [pkg] defined in [doc] of the given [dependencyOfRelation] type. Optionally, the
+     * [SpdxRelationship.Type.DEPENDS_ON] type is handled by [dependsOnCase].
+     */
+    private fun getDependencies(
+        pkg: SpdxPackage,
+        doc: SpdxDocument,
+        dependencyOfRelation: SpdxRelationship.Type,
+        dependsOnCase: (String) -> PackageReference? = { null }
+    ): SortedSet<PackageReference> =
+        doc.relationships.mapNotNullTo(sortedSetOf()) { (source, target, relation) ->
+            when {
+                // Dependencies can either be defined on the target...
+                target == pkg.spdxId && relation == dependencyOfRelation -> {
+                    val dependency = doc.packages.single { it.spdxId == source }
+                    PackageReference(
+                        id = dependency.toIdentifier(),
+                        dependencies = getDependencies(
+                            dependency,
+                            doc,
+                            SpdxRelationship.Type.DEPENDENCY_OF,
+                            dependsOnCase
+                        )
+                    )
+                }
+
+                // ...or on the source.
+                source == pkg.spdxId && relation == SpdxRelationship.Type.DEPENDS_ON -> {
+                    dependsOnCase(target)
+                }
+
+                else -> null
+            }
+        }
+
+    /**
+     * Return a [Scope] created from the given type of [relation] for [projectPackage] in [spdxDocument], or `null` if
+     * there are no such relations.
+     */
+    private fun createScope(spdxDocument: SpdxDocument, projectPackage: SpdxPackage, relation: SpdxRelationship.Type) =
```

Comment:
I believe this fun deserves an explicit return type.

Reply:
Are you saying it's not clear that a function called "createScope" creates a `Scope`?

Reply:
Yes

Reply:
Well, that honestly leaves me speechless, esp. as the KDoc already documents the return type. But I don't have the time to argue about it, and as I depend on your review I've made that change.

Reply:
it's optional - do whatever you like